### ISSUE TITLE
`cupy.diag` does not raise error on >2-dim input

### ISSUE THREAD

**Issue by toslunar** [cupy/cupy#1304]
```
>>> cupy.diag(cupy.ones((3,3,3)))
array([[1., 1., 1.],
       [1., 1., 1.],
       [1., 1., 1.]])
>>> numpy.diag(numpy.ones((3,3,3)))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/tos/GitHub/numpy/numpy/lib/twodim_base.py", line 265, in diag
    raise ValueError("Input must be 1- or 2-d.")
ValueError: Input must be 1- or 2-d.
```

**Comment by aksub99** [comment]
@toslunar I think this issue has been resolved in #1337 .

**Comment by toslunar** [comment]
@aksub99 Thanks for your investigation!
